src/editor/codemirror/language-server/signatureHelp.ts

```diff
@@ -72,17 +77,22 @@ const triggerSignatureHelpRequest = async (view: EditorView): Promise<void> => {
     position: offsetToPosition(view.state.doc, pos),
   };
   try {
+    setTimeout(() => {
+      view.dispatch({
+        effects: [setSignatureHelpRequestPosition.of(pos)],
+      });
+    }, 0);
     const result = await client.connection.sendRequest(
       SignatureHelpRequest.type,
       params
     );
     view.dispatch({
-      effects: [setSignatureHelpEffect.of({ pos, result })],
+      effects: [setSignatureHelpResult.of(result)],
     });
   } catch (e) {
     logException(view.state, e, "signature-help");
     view.dispatch({
-      effects: [setSignatureHelpEffect.of({ pos, result: null })],
+      effects: [setSignatureHelpResult.of(null)],
     });
```

Comment:
The essence of the change is that we no longer assume this position is still valid when we hear back from Pyright.
Instead we stash it before the request and map it forward through subsequent changes as required.